go/metadata/capabilities/capabilities.go

```diff
@@ -143,12 +147,89 @@ func normalize(key string, value interface{}) interface{} {
 			out[k] = strings.ToLower(v.(string))
 			continue
 		}
+		if k == "noProxy" {
+			if vs, ok := v.(string); ok {
+				out["noProxy"] = strings.Split(vs, ",")
+				continue
+			}
+		}
 		out[k] = v
 	}
 
+	return key, out
+}
+
+// denormalizeW3C remove non-W3C capabilities.
+func denormalizeW3C(in map[string]interface{}) map[string]interface{} {
+	out := map[string]interface{}{}
+
+	for k, v := range in {
+		if k == "goog:chromeOptions" {
+			denormalized["chromeOptions"] = v
+		}
+		// extension capabilities
+		if strings.Contains(k, ":") {
+			out[k] = v
+			continue
+		}
+		for _, a := range w3cSupportedCapabilities {
+			if k == a {
+				out[k] = v
+				break
+			}
+		}
+	}
+
 	return out
 }
 
+func denormalizeJWP(caps map[string]interface{}) map[string]interface{} {
+	denormalized := map[string]interface{}{}
+
+	for k, v := range caps {
+		if k == "goog:chromeOptions" {
+			denormalized["chromeOptions"] = v
+		}
+
+		if k != "proxy" {
+			denormalized[k] = v
+			continue
+		}
+
+		proxy, ok := v.(map[string]interface{})
+		if !ok {
+			denormalized[k] = v
+			continue
+		}
+
+		denormalizedProxy := map[string]interface{}{}
```

Comment:
Can you also normalize the order of items in the proxy capability? I'm still getting the same error, and I think it's because alwaysMatch["proxy"] has proxyType, noProxy, and httpProxy whereas firstMatch["proxy"] has httpProxy, noProxy, and proxyType. The actual values associated with those parameters are consistent now.

Reply:
This was caused by a type difference.

Also, the chromeOptions logic was not ideal, so I have updated it to merge chromeOptions and goog:chromeOptions instead of just having one replace the other.